pwndbg/lib/memory.py

```diff
@@ -140,7 +140,13 @@ def permstr(self) -> str:
         )
 
     def __str__(self) -> str:
-        return f"{self.vaddr:#{2 + 2 * pwndbg.aglib.arch.ptrsize}x} {self.vaddr + self.memsz:#{2 + 2 * pwndbg.aglib.arch.ptrsize}x} {self.permstr} {self.memsz:8x} {self.offset:6x} {self.objfile or ''}"
+        from os.path import relpath
+        if pwndbg.config.vmmap_prefer_relpaths:
+            rel = relpath(self.objfile)
+            objfile = self.objfile if rel.startswith('../../') else rel
```

Comment:
What's the logic/reasoning here with `../../`? What are the cases for it etc?

Reply:
Python's `relpath` will use relative paths even for libc from the system. So we have longer paths (with no clear benefits) over the absolute paths.

Should I document it in the code ?

Reply:
```
In [8]: os.path.relpath('/usr/lib/a/b/c/libc.so')
Out[8]: '../../../../usr/lib/a/b/c/libc.so'
```

Oh.. that's what you mean.

What about changing the implementation then to this?:
```py
prefix = os.path.commonprefix((os.getcwd(), self.objfile))
objfile = self.objfile.lstrip(prefix) if prefix != "/" else self.objfile
```

Two notes:
* `lstrip` is probably not best bcoz we may strip prefix twice in theory? probably doesn't matter
* shall we use `os.path.commonprefix` or `os.path.commonpath`? ugh...

Reply:
It's okay, as long as we document it appropriately, since it is not so obvious why we do the `"../../"` check.